aiida_common_workflows/cli/launch.py

```diff
@@ -0,0 +1,80 @@
+# -*- coding: utf-8 -*-
+"""Commands to launch common workflows."""
+import functools
+
+import click
+
+from aiida.cmdline.params import types
+
+from aiida_common_workflows.plugins import get_workflow_entry_point_names, load_workflow_entry_point
+from .root import cmd_root
+from . import options
+from . import utils
+
+
+@cmd_root.group('launch')
+def cmd_launch():
+    """Launch a common workflow."""
+
+
+@cmd_launch.command('relax')
+@click.argument('plugin', type=types.LazyChoice(functools.partial(get_workflow_entry_point_names, 'relax', True)))
+@options.STRUCTURE(help='The structure to relax.')
+@options.PROTOCOL(type=click.Choice(['fast', 'moderate', 'precise']), default='fast')
+@options.RELAXATION_TYPE()
+@options.THRESHOLD_FORCES()
+@options.THRESHOLD_STRESS()
+@options.DAEMON()
+@click.option('--show-engines', is_flag=True, help='Show information on the required calculation engines.')
+def cmd_relax(plugin, structure, protocol, relaxation_type, threshold_forces, threshold_stress, daemon, show_engines):
+    """Relax a crystal structure using the common relax workflow for one of the existing plugin implementations.
+
+    The command will automatically try to find and load the codes that are required by the plugin workflow. If no code
+    is installed for at least one of the calculation engines, the command will fail. Use the `--show-engine` flag to
+    display the required calculation engines for the selected plugin workflow.
+    """
+    # pylint: disable=too-many-locals
+    from aiida.orm import QueryBuilder, Code
+
+    process_class = load_workflow_entry_point('relax', plugin)
+    generator = process_class.get_inputs_generator()
+
+    if not generator.is_valid_protocol(protocol):
+        protocols = generator.get_protocol_names()
+        process_class_name = process_class.__name__
+        message = f'`{protocol}` is not implemented by `{process_class_name}` workflow: choose one of {protocols}'
+        raise click.BadParameter(message, param_hint='protocol')
+
+    if show_engines:
+        for engine in generator.get_calc_types():
+            schema = generator.get_calc_type_schema(engine)
+            click.secho(engine, fg='red', bold=True)
+            click.echo('Required code plugin: {}'.format(schema['code_plugin']))
+            click.echo('Engine description:   {}'.format(schema['description']))
+
+        return
+
+    engines = {}
+
+    for engine in generator.get_calc_types():
+        schema = generator.get_calc_type_schema(engine)
+        engines[engine] = {
+            'options': {
+                'resources': {
+                    'num_machines': 1
+                },
+                'max_wallclock_seconds': 86400,
+            }
+        }
```

Comment:
Here the resources for the calculation are set. What if the user wants to use a different number of processors?  What if my code is on a remote machine that requires to specify an account?? At the moment there is no way to do that from command line. This means the command-line approach will be limited to situations where this set of decided resources will work. In your opinion, is it feasible to define a flag that accepts directly the "engine" dictionary? Sure it would be ugly but I don't see alternatives.

Reply:
I see the problem and it is one that I have considered often before. The problem with CLI's like this is that there will always be a tradeoff between its ease of use and its limitations. It is (practically) impossible to make everything configurable. If a user wants ultimate flexibility, I think in the end it is always best to just have your own launch script. This way you are not limited by the interface of the CLI. This is why I often stress that the CLI is for demonstration and testing purposes. It should be simple to use, and work for the most basic cases. As soon as you need something more specific and for actual production, just write a script (where you can of course use part of the code of the CLI as an example).

That being said, I think it might be possible to have an option that takes a dictionary as a string. For example we could literal evaluation using the `ast` module:
```
import ast
d = ast.literal_eval("{'a': 5}"
print(d['a'])
```
however, this is quite dirty I would say. Besides, writing the dictionary as a single flat string would also be tedious for the user. So I am not sure that we should add something like this.

As a compromise, maybe we could just add two options that allow a user to configure the number of machines and wallclock seconds. These options will have to accept multiple values each, in the case that there are more than one engines that need to be run by the workflow. It would look something like:
```
aiida-common-workflow launch relax vasp -w 86400 3600 -n 4 2
```
where the first engine would run on 4 machines for 86400 seconds and the second step would take 2 machines for one hour. Note that your use case where a specific account should be used, the user can already setup the computer or code to have that account set in the prepend text.

Reply:
Yes this is always the problem and I agree with you that the CLI can't cover everything. However, if we want to introduce a change, the number of machines and wallclock seconds are, in my opinion, less relevant than the possibility of passing the account name, especially if, in the future, we will manage to implement something about #13. Leave it like that is also fine, I just wanted to point out this limitation.